esda/lee.py

```diff
@@ -0,0 +1,143 @@
+import numpy
+from scipy import sparse
+from sklearn.base import BaseEstimator
+from sklearn import preprocessing
+from sklearn import utils
+from libpysal.weights import lag_spatial, WSP
+
+def _lee_from_R(x,y,W, local, zero_policy=None, NAOK=False):
+    from rpy2.robjects.packages import importr
+    from rpy2.robjects import r as R, numpy2ri
+    from rpy2.rinterface import NULL
+
+    zero_policy = NULL if zero_policy is None else zero_policy
+
+    spdep = importr('spdep')
+    
+    numpy2ri.activate()
+    W_matrix_R = numpy2ri.py2ro(W.sparse.toarray())
+    W_listw_R = spdep.mat2listw(W_matrix_R)
+    result = spdep.lee(x.flatten(), y.flatten(), W_listw_R,
+                       len(x.flatten()), zero_policy=zero_policy, NAOK=NAOK)
+    
+    if local:
+        result = numpy2ri.ri2py(result[1])
+    else:
+        result = numpy2ri.ri2py(result[0])
+    numpy2ri.deactivate()
+    return result
+
+def RSpatial_Pearson(x,y,W, zero_policy=None, NAOK=False):
+    return _lee_from_R(x,y,W, local=False, zero_policy=zero_policy, NAOK=NAOK)
+
+def RLocal_Spatial_Pearson(x,y,W, zero_policy=None, NAOK=False):
+    return _lee_from_R(x,y,W, local=True, zero_policy=zero_policy, NAOK=NAOK)
+
+class Spatial_Pearson(BaseEstimator):
+    def __init__(self, connectivity=None, permutations=999):
+        self.connectivity = connectivity
+        self.permutations = permutations
```

Comment:
Should there be type caster here just in case the user give `permutations` as something other than an integer? 

```python
self.permutations = int(permutations)
```